File: core/support/src/main/java/com/reservation/support/exception/BusinessException.java
package com.reservation.support.exception;

import org.springframework.http.HttpStatus;

public class BusinessException extends RuntimeException {
	private final ErrorCode errorCode;

	BusinessException(ErrorCode errorCode, String message) {
		super(message);
		this.errorCode = errorCode;
	}

	public ErrorCode errorCode() {
		return errorCode;
	}

	public HttpStatus status() {
		return errorCode.status();
	}
}

File: core/support/src/main/kotlin/com/reservation/support/exception/BusinessException.kt
package com.reservation.support.exception

import org.springframework.http.HttpStatus

class BusinessException internal constructor(private val errorCode: ErrorCode, message: String?) :
    RuntimeException(message) {
    fun errorCode(): ErrorCode {
        return errorCode
    }

    fun status(): HttpStatus {
        return errorCode.status()
    }
}
